res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -1907,41 +1875,19 @@ NumarkMixtrack3.BeatKnob = function(channel, control, value, status, group) {
         }
     }
 
-    if (deck.TapDown) {
-        engine.setValue('[BeatJump]', increase ? 'next': 'prev', true);
-        engine.setValue('[BeatJump]', increase ? 'next': 'prev', false);
-    }
-
     if (deck.shiftKey) {
-        engine.setValue('[LoopMove]', increase ? 'next': 'prev', true);
-        engine.setValue('[LoopMove]', increase ? 'next': 'prev', false);
-    }
-};
-
-NumarkMixtrack3.OnBeatJump = function(value, group, control) {
-    if (!value) return;
-    // function to use to determine next jump size depends on value of control
-    var nextSizeFun = (control === 'next') ? nextJumpSize : prevJumpSize;
-    beatJumpSize = nextSizeFun(beatJumpSize);
-};
-
-NumarkMixtrack3.OnLoopMove = function(value, group, control) {
-    if (!value) return;
-    var nextSizeFun = (control === 'next') ? nextJumpSize : prevJumpSize;
-    loopMoveSize = nextSizeFun(loopMoveSize);
-};
+        var presetJumpSizes = [1/32, 1/16, 1/8, 1/4, 1/2, 1, 2, 4, 8, 16, 32, 64, 128, 256, 512];
+        var jumpSize = engine.getValue(deck.group, 'beatjump_size');
+        var index = presetJumpSizes.indexOf(jumpSize);
 
-// when we hit a button on the UI that beat jumps by a certain amount, reset beatJumpSize
-// to reflect this. This mostly prevents the script and UI from becoming out of sync
-// when the script is reset but the UI is not.
-NumarkMixtrack3.OnBeatJumpX = function(value, group, control) {
-    if (!value) return;
-    beatJumpSize = parseInt(control.match(/\d+/)[0]);
-};
+        if (increase) {
+            jumpSize = presetJumpSizes[(index + 1) % (presetJumpSizes.length)];
```

Comment:
Is there a need for wrapping around when reaching the smallest/biggest value? When I was developing #1187 I found such behavior to be a little unexpected. I think checking against a minimum and maximum value is sufficient. This would also allow the user to enter a non-power-of-2 number into the spinbox and halve/double it from the controller.